src/android/com/adobe/phonegap/push/PushPlugin.java

```diff
@@ -119,10 +119,13 @@ else if (!savedSenderID.equals(senderID)) {
                         editor.commit();
                     }
 
-                    if (gCachedExtras != null) {
+                    if (!gCachedExtras.isEmpty()) {
                         Log.v(LOG_TAG, "sending cached extras");
-                        sendExtras(gCachedExtras);
-                        gCachedExtras = null;
+                        Iterator<Bundle> gCachedExtrasIterator = gCachedExtras.iterator();
+                        while (gCachedExtrasIterator.hasNext()) {
+                            sendExtras(gCachedExtrasIterator.next());
```

Comment:
I think it is cleaner and targeting this for a 1.5.0 release anyway so I'm okay with making that change.